lib/plugin.js

```diff
@@ -1,21 +1,31 @@
 import Vue from 'vue'
 import Segment from '@dansmaculotte/vue-segment'
 
-const SEGMENT_WRITE_KEY = '<%= options.writeKey %>'
-const SEGMENT_DISABLED = <%= options.disabled %>
-const SEGMENT_USE_ROUTER = <%= options.useRouter %>
-const SEGMENT_SETTINGS = <%= JSON.stringify(options.settings) %>
+function checkIfBoolean (param) {
+  return typeof param === 'boolean'
+}
+
+const OPTIONS = '<%= JSON.stringify(options) %>'
 
 export default function (context, inject) {
-  const { app, store } = context
+  const { app, store, $config } = context
+
+  const moduleOptions = JSON.parse(OPTIONS)
 
   const options = {
-    writeKey: SEGMENT_WRITE_KEY,
-    disabled: SEGMENT_DISABLED,
-    settings: SEGMENT_SETTINGS
+    writeKey: moduleOptions.writeKey || ($config && $config.SEGMENT_WRITE_KEY),
+    disabled: moduleOptions.disabled || ($config && $config.SEGMENT_DISABLED) || false,
+    settings: moduleOptions.settings,
   }
 
-  if (SEGMENT_USE_ROUTER && app.router) {
+  const useRouter =
```

Comment:
Can you hide this chain of ternaries behind a function? I think it will be more readable.